appium/webdriver/webdriver.py

```diff
@@ -142,6 +148,40 @@ def __init__(self, command_executor='http://127.0.0.1:4444/wd/hub',
         By.IMAGE = MobileBy.IMAGE
         By.CUSTOM = MobileBy.CUSTOM
 
+    def _update_command_executor(self, keep_alive):
+        """Update command executor following directConnect feature"""
+        direct_protocol = 'directConnectProtocol'
+        direct_host = 'directConnectHost'
+        direct_port = 'directConnectPort'
+        direct_path = 'directConnectPath'
+
+        if (not {direct_protocol, direct_host, direct_port, direct_path}.issubset(set(self.capabilities))):
+            message = 'Direct connect capabilities from server were:\n'\
+                '{protocol}: {protocol_get},  {host}: {host_get}, {port}: {port_get}, {path}: {path_get}'.format(
+                    protocol=direct_protocol, protocol_get=self.capabilities.get(direct_protocol, ''),
```

Comment:
capabilities is already a dictionary, so we could in theory to something like `**self.capabilities`

Reply:
I'd like to show a blank in no key case like below, so I changed to `for` style

```
directConnectProtocol: 'http'
directConnectHost: 'localhost2'
directConnectPort: '4800'
directConnectPath: ''
```